server/server.go

```diff
@@ -27,38 +27,42 @@ func New(db db.SQLClient, authority auth.Authenticator, cf cloudfoundry.Client)
 	// Never crash on panic
 	e.Use(middleware.Recover())
 
-	// Require a token for all requests
-	e.Use(auth.UAATokenAuthentication(authority))
-
 	// Validate and parse range query param
 	e.Use(api.ValidateRangeParams)
 
+	authGroup := e.Group("", auth.UAATokenAuthentication(authority))
+	adminGroup := authGroup.Group("", auth.AdminOnly)
+
+	e.GET("/oauth/authorize", authority.Authorize)
+	e.GET("/oauth/callback", authority.Exchange)
+
+	e.POST("/report/:org_guid", api.NewSimulatedReportHandler(db))
```

Comment:
One last thing 😇 

I'm not a fan of the name clash between the `POST /report/:org_guid` route and the `GET /report/:org_guid`.

I think we should prefix our 'estimatation' endpoints with something to make it clear that it is a unique feature to do with predicting costs and we can probably safely drop the :org_guid as it doesn't add much value.

I think `e.POST("/forcast/report", NewSimulatedReportHandler)` would be good.

Reply:
Yes sir!